src/dao/AbstractContractDAO.js

```diff
@@ -400,7 +399,9 @@ export default class AbstractContractDAO {
             } catch (e) {
               errorCode = txErrorCodes.FRONTEND_UNKNOWN
             }
-            throw new TxError('Error event was emitted', errorCode)
+            if (!this._txOkCodes.includes(errorCode)) {
```

Comment:
Error event should not contain ok code
If this error was caught, then we should fix contracts code
So remove this if back